<?xml version="1.0"?>
<launch>

    <!-- launch your sensor driver here -->
    <include file="$(find hfl_driver)/launch/hfl118_node.launch"/>
    
    <param name="scan_lines" value="32" />
    <param name="period" value="0.1" />

    <param name="vertical_angle" type="double" value="2.0" />
    <param name="map_resolution" type="double" value="0.2" />
    <param name="max_dis" type="double" value="90.0" />
    <param name="min_dis" type="double" value="0.5" />

    <node pkg="floam" type="floam_odom_node" name="floam_odom_node" output="screen"/>
    <node pkg="floam" type="floam_lidar_node" name="floam_lidar_node" output="screen"/>
    <node pkg="floam" type="floam_mapping_node" name="floam_mapping_node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="word2map_tf"  args="0 0 0 0 0 0 /world /map 10" />
    
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find floam)/rviz/floam_velodyne.rviz" />
    </group>

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_loam" ns="base_link" >
        <param name="/target_frame_name" value="world" />
        <param name="/source_frame_name" value="base_link" />
        <param name="/trajectory_update_rate" value="10.0" />
        <param name="/trajectory_publish_rate" value="10.0" />
    </node>

</launch>
